config/confighttp/confighttp.go

```diff
@@ -124,6 +128,16 @@ func (hcs *HTTPClientSettings) ToClient(host component.Host, settings component.
 		transport.IdleConnTimeout = *hcs.IdleConnTimeout
 	}
 
+	// Setting the Proxy URL
+	if hcs.Proxy != "" {
+		proxyURL, err := url.Parse(hcs.Proxy)
+		if err != nil {
+			return nil, err
+		} else {
+			transport.Proxy = http.ProxyURL(proxyURL)
+		}
+	}
+
```

Comment:
Looks good, but needs a new test.

Reply:
Yes, I am working on the tests but facing a bit of issue configuring parameters. I'll ping you as soon as I have them :)